app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -585,33 +566,45 @@ class HomeActivityTest {
     profileTestHelper.loginToNewUser()
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
-      onView(withId(R.id.home_recycler_view)).perform(
-        scrollToPosition<RecyclerView.ViewHolder>(2)
-      )
-      onView(
-        atPositionOnView(
-          R.id.home_recycler_view,
-          2,
-          R.id.all_topics_text_view
-        )
-      ).check(
-        matches(withText(R.string.all_topics))
+      scrollToPosition(position = 2)
+      verifyExactTextOnHomeListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.all_topics_text_view,
+        stringToMatch = context.getString((R.string.all_topics))
       )
     }
   }
 
   @Test
-  fun testHomeActivity_noTopicsStarted_displaysAllTopicCards() {
+  fun testHomeActivity_noTopicsStarted_displaysAllTopicCardsInPortrait() {
     // Only new users will have no progress for any topics.
     profileTestHelper.loginToNewUser()
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
-      onView(withId(R.id.home_recycler_view)).perform(
-        scrollToPosition<RecyclerView.ViewHolder>(3)
+      scrollToPosition(position = 3)
+      onView(withId(R.id.home_recycler_view)).check(
+        // The "All Topics" section currently should display all four test topics in two rows.
+        hasGridColumnCount(expectedColumnCount = 2)
       )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_noTopicsStarted_displaysAllTopicCardsInLandscape() {
+    // Only new users will have no progress for any topics.
+    profileTestHelper.loginToNewUser()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(position = 3)
+      var rowsUsed = 2
+      if (context.resources.getBoolean(R.bool.isTablet)) {
+        // A landscape tablet can display the four test topics in one row.
+        rowsUsed = 1
+      }
```

Comment:
We should avoid control structures in tests because it makes the test harder to verify as correct. Instead of checking whether we're on tablet, let's have two versions of the test: one non-tablet and one tablet where we know exactly how many columns to expect for each.

Reply:
Done

Added different cases for mobile port / mobile land / tablet port / tablet land to assert the correct column count and checks the span of the topic cards